web/controllers/password_reset_controller.ex

```diff
@@ -11,14 +11,14 @@ defmodule CodeCorps.PasswordResetController do
   and return email. 422 if pwd do not match or auth token does not exist
   """
   def reset_password(conn, %{"token" => token, "password" => password, "password_confirmation" => password_confirmation}) do
-    user = conn.assigns.current_user
-    with %AuthToken{value: auth_token} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token, user_id: user.id }),
+    with %AuthToken{value: auth_token, user_id: user_id} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token }),
       {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", auth_token, max_age: 1209600) do
-        with %Changeset{valid?: true} <- User.reset_password_changeset(user, 
-                                                                       %{password: password, password_confirmation: password_confirmation}) do
+        user = Repo.get(User, user_id)
+        with %Changeset{valid?: true} = changeset <- User.reset_password_changeset(user, 
+                                                                       %{password: password, password_confirmation: password_confirmation}),
+             {:ok, user} <- Repo.update(changeset) do
           conn
-          |> put_status(:created)
-          |> render("show.json", email: user.email)
+          |> TokenController.create(%{"username" => user.email, "password" => password})
```

Comment:
Nice!  Yeah muchhh better.  Thanks a ton for that!  

One thing is this was the error I got for `render(:errors, data: changeset)` so I did `render(CodeCorps.ErrorView, "422.json")`

![image](https://cloud.githubusercontent.com/assets/7374640/25879713/074c65b4-34e8-11e7-97a3-7a9adff35213.png)